src/prefix-tree/node.ts

```diff
@@ -1,13 +1,18 @@
 import type { Nullable } from "../types";
 import { uniqueId } from "../utils";
 
+interface DocRef {
+  id: string;
+  tf: number;
+}
+
 export interface Node {
   id: string;
   key: string;
   word: string;
   parent: Nullable<string>;
   children: Record<string, string>;
-  docs: string[];
+  docs: DocRef[];
```

Comment:
Instead of an array of objects, what about a `Map<docId, tf>`? It will make all the calls O(1) instead of O(N)

Reply:
You can use a plain object for that using the key as `docId`. Example:

```js
{
  "1321321-2": 3
}
```

In which step the serialization happens?

Reply:
Yes that's another way I'd like to try! I think I'll make some benchmarks to decide how to proceed. I'm quite sure linked lists will perform slower when compared to iterating over object keys, but I wanna make sure.

The only real problem with the object approach is that every key can only have one value, and we may want to add more metadata in the future

Reply:
Serialization happens via https://github.com/lyrasearch/plugin-data-persistence, as it's a runtime-specific procedure (especially when you write on disk)